pwa-devdocs/src/tutorials/pwa-studio-fundamentals/manage-state/index.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Manage component and application state
+---
+
+Your components can respond to dynamic events in your application by keeping track of internal or global state.
+This tutorial provides the steps for creating a button that manages its own internal state and updating it to use data from the global application state.
+
+By the end of this tutorial, you will know how to add internal state to a component and use Peregrine's context hooks to access global state data.
+
+For more information on state management, read the [State management][] topic.
+
+{: .bs-callout .bs-callout-info}
+This tutorial requires you to have a project set up using the steps provided in the [Project Setup][] tutorial.
+
+## Create a new component
+
+For this part of the tutorial, you will create a React functional component called _MyComponent_.
+This component will render a button that toggles and displays an internal boolean state.
+
+### Create the component directory
+
+Create a new `MyComponent` directory under your project's `src/components` directory.
+
+```sh
+mkdir -p src/components/MyComponent
+```
+
+### Create the component
+
+Under `src/components/MyComponent`, create a `myComponent.js` file with the following content:
+
+```jsx
+import React, { useState, useCallback } from 'react';
+
+import Button from "@magento/venia-ui/lib/components/Button";
+
+const MyComponent = () => {
+    const [booleanStatus, setBooleanStatus] = useState(false);
+
+    const toggleStatus = useCallback(() => {
+        setBooleanStatus(!booleanStatus);
+    }, [booleanStatus]);
```

Comment:
So this'll work for this example but the best practice for updating a `useState` value based on its previous value is to use the [functional form of useState](https://reactjs.org/docs/hooks-reference.html#functional-updates).

Long story short, best practice here would be 
```
setBooleanStatus(previousStatus => !previousStatus);
```

This would actually also allow you to remove `booleanStatus` from the dependency array of `useCallback` too.